src/containers/ModelPanel/index.tsx

```diff
@@ -62,71 +64,67 @@ interface ModelPanelProps {
-                isSharedCheckboxIndeterminate={isSharedCheckboxIndeterminate}
-                recentColors={recentColors}
-                setColorChange={setColorChange}
-                setRecentColors={setRecentColors}
-            />
-        );
-        const contentMap = {
-            [ViewerStatus.Success]: checkboxTree,
-            [ViewerStatus.Empty]: (
-                <NoTrajectoriesText selectFile={loadNetworkFile} />
-            ),
-            [ViewerStatus.Loading]: <div />,
-            [ViewerStatus.Error]: isNetworkedFile ? (
-                <NetworkFileFailedText />
-            ) : (
-                <NoTypeMappingText />
-            ),
-        };
+const ModelPanel: React.FC<ModelPanelProps> = ({
+    uiDisplayDataTree,
+    agentHighlightMap,
+    agentVisibilityMap,
+    turnAgentsOnByDisplayKey,
+    highlightAgentsByDisplayKey,
+    setAgentsVisible,
+    payloadForSelectAll,
+    payloadForSelectNone,
+    isSharedCheckboxIndeterminate,
+    viewerStatus,
+    isNetworkedFile,
+    changeToNetworkedFile: loadNetworkFile,
+    recentColors,
+    setColorChange,
+    setRecentColors,
+    selectedAgent,
+}): JSX.Element => {
+    const checkboxTree = (
+        <CheckBoxTree
+            treeData={uiDisplayDataTree}
+            handleAgentCheck={turnAgentsOnByDisplayKey}
+            agentsChecked={agentVisibilityMap}
+            handleHighlight={highlightAgentsByDisplayKey}
+            agentsHighlighted={agentHighlightMap}
+            setAgentsVisible={setAgentsVisible}
+            payloadForSelectAll={payloadForSelectAll}
+            payloadForSelectNone={payloadForSelectNone}
+            isSharedCheckboxIndeterminate={isSharedCheckboxIndeterminate}
+            recentColors={recentColors}
+            setColorChange={setColorChange}
+            setRecentColors={setRecentColors}
+        />
+    );
+    const contentMap = {
+        [ViewerStatus.Success]: checkboxTree,
+        [ViewerStatus.Empty]: (
+            <NoTrajectoriesText selectFile={loadNetworkFile} />
+        ),
+        [ViewerStatus.Loading]: <div />,
+        [ViewerStatus.Error]: isNetworkedFile ? (
+            <NetworkFileFailedText />
+        ) : (
+            <NoTypeMappingText />
+        ),
+    };
 
-        return (
-            <div className={styles.container}>
-                <SideBarContents
-                    mainTitle="Agents"
-                    content={[
-                        <div className={styles.container} key="molecules">
-                            {contentMap[viewerStatus]}
-                        </div>,
-                        null,
-                    ]}
-                />
-            </div>
-        );
-    }
-}
+    return (
+        <div className={styles.container}>
+            <SideBarContents
+                mainTitle="Agents"
+                content={[contentMap[viewerStatus]]}
+                selectedAgent={selectedAgent}
+                uidisplayData={uiDisplayDataTree}
```

Comment:
```suggestion
                uiDisplayData={uiDisplayDataTree}
```
looks like the d should be uppercased for consistency.